.changeset/long-crabs-change.md

```diff
@@ -0,0 +1,5 @@
+---
+'@hashicorp/react-text-input': minor
+---
+
+- ✨ Adds support for a className prop.
```

Comment:
Adding `className` support to `text-input` was necessary to avoid a `:global` override selector previously used in `hero`.